crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs

```diff
@@ -89,14 +88,68 @@ impl SenderData {
         }
     }
 
+    /// Create a [`SenderData`] which contains device info.
+    pub fn device_info(device_keys: DeviceKeys) -> Self {
+        Self::DeviceInfo {
+            device_keys,
+            // TODO: when we have implemented all of SenderDataFinder,
+            // legacy_session should be set to false, but for now we leave
+            // it as true because we might lose device info while
+            // this code is still in transition.
+            legacy_session: true,
+        }
+    }
+
+    /// Create a [`SenderData`] with a known sender.
+    pub fn sender_known(
+        user_id: &UserId,
+        device_id: &DeviceId,
+        master_key: Ed25519PublicKey,
+        master_key_verified: bool,
+    ) -> Self {
+        Self::SenderKnown {
+            user_id: user_id.to_owned(),
+            device_id: Some(device_id.to_owned()),
+            master_key: Box::new(master_key),
+            master_key_verified,
```

Comment:
I'm starting to ponder if this is not quite as good of a name as we could have. We're calling this `master_key_verified` which would imply that two things need to be true for this field to be true:

1. The master key of the user identity has been signed by our own identity.
2. Our own identity is trusted.

But the finder thing also checks if the device is signed by the users identity as well, so it checks the whole chain `own identity -> other identity -> device` and only if the whole chain is valid and trusted then we set this to true.

Reply:
Any suggestions for a better name? @BillCarsonFr what do you think?